### PR TITLE
docs: Eventstore architecture

### PR DESCRIPTION
**Pull request by hifabienne**

### CODE REVIEW

File: docs/docs/concepts/eventstore/overview.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+## What is an eventstore?
+
+Traditionally, data is stored in relations as a state
+
+- A request needs to know the relations to select valid data
+- If a relation changes, the requests need to change as well
+- That is valid for actual, as well as for historical data
+
+An Eventstore on the other hand stores events, meaning every change that happens to any piece of data relates to an event.
+The data is stored as events in an append-only lof.
```
Comment (eliobischof) [inline docs/docs/concepts/eventstore/overview.md:16]:
```suggestion
The data is stored as events in an append-only log.
```

File: docs/docs/concepts/eventstore/overview.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+- Think of it as a ledger that gets new entries over time, accumulative
+- To request data, all you have to do is to sum the events as the summary reflects the actual state
+- To investigate past changes to your system, you just select the events from your time range of interest
+- That makes audit/analytics very powerful, due to the historical data available to build queries
+
+## Benefits
+
+Audit: You have a built-in audit trail that tracks all changes over an unlimited period of time.
+Travel back in time: With our way of storing data we can show you all of your resources at a given point in time.
```
Comment (eliobischof) [inline docs/docs/concepts/eventstore/overview.md:26]: How to do that when troubleshooting?

File: docs/docs/concepts/eventstore/overview.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+- That makes audit/analytics very powerful, due to the historical data available to build queries
+
+## Benefits
+
+Audit: You have a built-in audit trail that tracks all changes over an unlimited period of time.
+Travel back in time: With our way of storing data we can show you all of your resources at a given point in time. 
+
+## Definitions
+
+Eventsourcing has some specific terms that are often used in our documentation. To unserstand how ZITADEL works it is important to understand this key definitions.
```
Comment (eliobischof) [inline docs/docs/concepts/eventstore/overview.md:30]:
```suggestion
Eventsourcing has some specific terms that are often used in our documentation. To understand how ZITADEL works it is important to understand this key definitions.
```

File: docs/docs/concepts/eventstore/overview.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+### Events
+
+An event is something that happen in the system and got written to the database. This is the single source of truth.
```
Comment (eliobischof) [inline docs/docs/concepts/eventstore/overview.md:34]:
```suggestion
An event is something that happens in the system and gets written to the database. This is the single source of truth.
```

File: docs/docs/concepts/eventstore/overview.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+Events are immutable and the current state of your system is derived from the events.
+
+Possible Events:
+- user.added
+- user.changed
+- product.added
+- user.password.checked
+
+### Aggregate
```
Comment (eliobischof) [inline docs/docs/concepts/eventstore/overview.md:43]: Are they stored in a write model?

File: docs/docs/concepts/eventstore/zitadel.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+| id | unique identifier of the event, this is generated by the database  | b6402a60-e4655-4cc0-904b-f5c6760c4406 |
+| aggregate_type | the type of the aggregate, an aggregate can be compared to a resource or an object. One aggregate contains multiple event types | user |
+| aggregate_id | The unique identifier of an aggregate, this is generated by ZITADEL and is a sonyflake id | 168096909691353697 |
+| aggregate_version | The aggregate version shows in which version of the aggregate the event was created. This is needed to be able to compute correct objects | v1 |
+| event_type | The type of the event | user.human.added |
+| event_sequence | The event sequence is a sequence number that is incremented by one for each event on the instance. For technical reasons, a number can be omitted in some cases. This is needed so that the sequence of the events can be ensured. | 1234 | 
+| previous_aggregate_sequence| This number is the sequence of the event last created on this specific aggregate. E.g. Last user with specific aggregate_id | 1233|
+| previous_aggregate_type_sequence | This number is the sequence of the event last created on this aggregate. E.g Last User | 1230 |
```
Comment (eliobischof) [inline docs/docs/concepts/eventstore/zitadel.md:24]: What are they used for? Sanity checks?

File: docs/docs/concepts/eventstore/zitadel.md [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+| resource_owner | The resource owner defines to which organization/resource_owner the event belongs. This is an id generated by ZITADEL as sonyflake id | 168051083313153168 |
+| instance_id | ZITADEL is capable of containing multiple ZITADEL instances withing the system. This id is the unique identifier of the Instance and is generated by ZITADEL as sonyflake id. | 165460784409737865 |
+
+
+## Schemas
+
+| Schema | Description | Examples |
+| --- | --- | --- |
+| System | The system contains everything that is needed outside the ZITADEL instances.  | assets, encryption_key |
+| Eventstore | Eventstore is the base of ZITADEL and is the single source of truth. All the events stored in the eventstore can be used in to generated different projections. | events, instance sequences, system wide unique constraints
```
Comment (eliobischof) [inline docs/docs/concepts/eventstore/zitadel.md:37]:
```suggestion
| Eventstore | Eventstore is the base of ZITADEL and is the single source of truth. All the events stored in the eventstore can be used to generate different projections. | events, instance sequences, system wide unique constraints
```

File: docs/docs/concepts/eventstore/zitadel.md [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+
+You can find the failed_events in the following tables:
+- projections.failed_events
+- notification.failed_events
+- auth.failed_events
+- adminapi.failed_events
+
+| Attribute | Description | Examples |
+| --- | --- | --- |
+| projection_name | The name of the projection for which the failed even should have been processed.  | projection.users |
```
Comment (eliobischof) [inline docs/docs/concepts/eventstore/zitadel.md:90]:
```suggestion
| projection_name | The name of the projection for which the failed event should have been processed.  | projection.users |
```

**Review by eliobischof — CHANGES_REQUESTED** [review]

File: docs/docs/concepts/eventstore/overview.md (hunk 5 above)
Comment (hifabienne) [reply docs/docs/concepts/eventstore/overview.md:43]: No an aggregate is actually just all events from a specific "object" together. The write model is used in the command side to write something to the eventstore.

Comment (eliobischof) [reply docs/docs/concepts/eventstore/overview.md:43]: So I understand it is computed and stored in-memory and can't be inspected in the db.

Comment (eliobischof) [reply docs/docs/concepts/eventstore/overview.md:43]: Maybe we can write that for clarity?

File: docs/docs/concepts/eventstore/zitadel.md (hunk 6 above)
Comment (adlerhurst) [reply docs/docs/concepts/eventstore/zitadel.md:24]: they are used for the projections. They compare if the current sequence of the projection matches this sequence

**Review by eliobischof — APPROVED** [review]

**Pull request merged**